keras/layers/preprocessing/discretization_test.py

```diff
@@ -124,5 +125,11 @@ def test_saving(self):
         self.assertAllClose(layer(ref_input), ref_output)
 
     def test_sparse_inputs(self):
-        # TODO
-        pass
+        if backend.backend() != "tensorflow":
```

Comment:
You should use a pytest decorator for this (we do this in a number of places elsewhere, check it out)

Reply:
Thanks for the help, changed it to the decorator.

One small comment, when running

`sh shell/format.sh`

I had to change line `isort --sp "${base_dir}/pyproject.toml" ` to `isort -sp "${base_dir}/pyproject.toml" ` because I was getting this error:

usage: isort [-h] [-a ADD_IMPORTS] [-ac] [-af] [-b KNOWN_STANDARD_LIBRARY]
             [-c] [-ca] [-cs] [-d] [-df] [-ds] [-dt] [-e]
             [-f KNOWN_FUTURE_LIBRARY] [-fas] [-fass] [-ff FROM_FIRST]
             [-fgw [FORCE_GRID_WRAP]] [-fss] [-i INDENT] [-j JOBS] [-k]
             [-l LINE_LENGTH] [-lai LINES_AFTER_IMPORTS]
             [-lbt LINES_BETWEEN_TYPES] [-le LINE_ENDING] [-ls]
             [-m {0,1,2,3,4,5,6,7}] [-nis] [-nlb NO_LINES_BEFORE]
             [-ns NOT_SKIP] [-o KNOWN_THIRD_PARTY] [-ot]
             [-p KNOWN_FIRST_PARTY] [-q] [-r] [-rm REMOVE_IMPORTS] [-rr] [-rc]
             [-s SKIP] [-sd DEFAULT_SECTION] [-sg SKIP_GLOB] [-sl]
             [-sp SETTINGS_PATH] [-t FORCE_TO_TOP] [-tc] [-up] [-v] [-vb]
             [--virtual-env VIRTUAL_ENV] [--conda-env CONDA_ENV] [-vn]
             [-w LINE_LENGTH] [-wl WRAP_LENGTH] [-ws] [-y] [--unsafe]
             [--case-sensitive] [--filter-files]
             [files [files ...]]
isort: error: unrecognized arguments: --sp .

I am running the script using git bash

Reply:
I was indeed running an old version. I updated it and run it, the imports should now be correctly sorted :)